pkg/operator/cluster.go

```diff
@@ -77,23 +93,50 @@ func (t *clusterTPR) findAllClusters() error {
 	logger.Infof("found %d clusters", len(clusterList.Items))
 	for i := range clusterList.Items {
 		c := clusterList.Items[i]
+		t.startCluster(&c)
+	}
 
-		ns := c.Spec.Namespace
-		existingCluster := cluster.New(c.Spec, t.context.factory, t.context.clientset)
-		t.tracker.add(ns, c.Metadata.ResourceVersion)
-		t.clusters[ns] = existingCluster
+	t.watchVersion = clusterList.Metadata.ResourceVersion
+	return nil
+}
 
-		logger.Infof("resuming cluster %s in namespace %s", c.Metadata.Name, ns)
-		t.startCluster(existingCluster)
+func (t *clusterManager) startTrack(c *cluster.Cluster) error {
+	t.Lock()
+	defer t.Unlock()
+
+	existing, ok := t.clusters[c.Namespace]
+	if ok {
+		if c.Name != existing.Name {
+			return fmt.Errorf("cluster %s already running in namespace %s. multiple clusters per namespace not supported", existing.Name, existing.Namespace)
```

Comment:
cluster %s `is` already running. `M`ultiple clusters per namespace not supported`.`"